pkg/utils/utils.go

```diff
@@ -330,3 +330,24 @@ func StringSliceEqual(s1, s2 []string) bool {
 
 	return true
 }
+
+// MergeMap merges the m2 into m1, if it has the same keys, m2 will overwrite m1.
+func MergeMap(m1 map[string]interface{}, m2 map[string]interface{}) (map[string]interface{}, error) {
+	if m1 == nil && m2 == nil {
+		return nil, fmt.Errorf("all of maps are nil")
```

Comment:
Why does it return an error? I think it is quite normal to merge two nil map.

I think it is incorrect to return err. Instead, the upper caller could judge whether the returned value is nil to return error or something.

`MergeMap` is a general function and fundamental, it should not add some business logic in the implementation. For me, judging both of them nil is business logic.